build_scripts/build.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+# The MIT License (MIT)
+
+# Copyright 2022 Hewlett Packard Enterprise Development LP.
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+VENV_PATH="/tmp/poetryvenv"
+
+function in_venv() {
+    if ! [[ -d ${VENV_PATH} ]]; then
+        /usr/bin/python3 -m venv "$VENV_PATH"
+    fi
+
+    # Execute the command directly within the venv
+    "${VENV_PATH}/bin/$@"
```

Comment:
Part of the issue might be that you need to run `./build_scripts/build.sh prep` first? The dependency isn't explicit unfortunately but it works in Jenkins since the stages are run in order each time. I considered using a Makefile like you did in `shasta-install-utility-common` (e.g. [here](https://github.com/Cray-HPE/shasta-install-utility-common/blob/main/Makefile) ) but I wasn't sure how to properly encode the dependencies for creating the virtualenv in the first place (e.g. how would `make` know if the virtualenv had been created for the project already? There's probably a way but `make` gets a bit unwieldy at a certain complexity level.)

Reply:
I get the error even when running `prep` first, but I think I might be missing something.

Overall it also seems to all work if I just create the virtualenv first, e.g.
```
$ virtualenv /tmp/poetryvenv/
$ source /tmp/poetryvenv/bin/activate
$ ./build_scripts/build.sh prep
$ ./build_scripts/build.sh unittest
```

Reply:
Just to quickly follow up on this, this wasn't *all* the dependencies, it was just a couple of them. I think I am having some issue on my end here but I don't think is anything to do with poetry or what's in this PR.